docs(dev): add documentation for developer (#322)
SocialGouv/cdtn-admin: README-dev.md

# Code du travail numérique - Administration - dev

## URLs

| Environnement                                              | URL                                                         |
|------------------------------------------------------------|-------------------------------------------------------------|
|    Production (access granted only for authorized IP)      |  <https://cdtn-admin.fabrique.social.gouv.fr/>              |

## Overview

> This git is a monorepo composed of 4 projects.

Reviewer: ```suggestion
This git repository is a monorepo composed of 4 projects.
```

### Hasura

Used to expose data through a GraphQL API.
It contains the metadata and migrations for Hasura.

See the [Hasura documentation](https://hasura.io/docs/1.0/graphql/core/index.html) for more information.

> 

Reviewer: ```suggestion
We recommend to [install the hasura console](https://hasura.io/docs/1.0/graphql/core/hasura-cli/install-hasura-cli.html) which provides a graphql sandbox and an administration UI for hasura 
` ``

### Ingester

Used to populate the database with documents provided by external sources.

There is, at this moment (February 2021), 4 sources:
 * [@SocialGouv/contributions-data](https://github.com/SocialGouv/contributions-data)
 * [@SocialGouv/fiches-travail-data](https://github.com/SocialGouv/fiches-travail-data)
 * [@SocialGouv/fiches-vdd](https://github.com/SocialGouv/fiches-vdd)
 * [@SocialGouv/kali-data](https://github.com/SocialGouv/kali-data)
 * [@SocialGouv/legi-data](https://github.com/SocialGouv/legi-data)

> Each github repo has released. In the release, we have the content as json. 

Reviewer: ```suggestion
Each github repo use releases to track changes. Each release expose content as json. 
```

Author: ```suggestion
Each GitHub repo uses releases to track changes. Each release exposes content as JSON.
```

For each new release of an external packages, this script compares the content and insert diff in the database.

See [documentation](targets/alert-cli/README.md) for more detail.

## Setup

Make sure you're using NodeJS 12+.

```sh
# Install all the packages
$ yarn
$ yarn build
```

It's easy to setup a new environment with docker compose :

```sh
docker-compose up
```

The docker compose performs several steps.

### Configure a postgreSQL database

A postgreSQL database is used to store the data exposed through a Hasura instance.

> Start only the postgreSQL instance:
> ```sh
> docker-compose up postgres
> ```

### Configure a Hasura instance

A Hasura instance is used to expose the data stored in postgreSQL through a GraphQL API. 
See the [Hasura documentation](https://hasura.io/docs/1.0/graphql/core/index.html) for more information.

This step creates a new Hasura instance with the schema, 
and some data (see [metadata](targets/hasura/metadata) and [migrations](targets/hasura/migrations) files of hasura target).

To access to the Hasura console, run this command:

```sh
hasura console --envfile ../../.env --project targets/hasura
```

> A webpage is opened in your browser. The password is `admin1`.

Reviewer: ```suggestion
A webpage is opened in your browser. The password is `admin1` as set in the `.env` file.
```

Author: ```suggestion
A webpage is opened in your browser. The password is `admin1` as set in the `.env` file (`HASURA_GRAPHQL_ADMIN_SECRET` key).
```
